search-app/src/cmr/search/site/data.clj

```diff
@@ -79,31 +80,49 @@
-                                  :result-features [:granule-counts]
-                                  :result-fields [:concept-id
-                                                  :doi
-                                                  :entry-title
-                                                  :short-name
-                                                  :version-id]})
-        result (query-exec/execute-query context query)
-        {:keys [items granule-counts-map]} result]
-    (sort-by :entry-title
-             (map #(assoc % :granule-count (get granule-counts-map (:concept-id %))) items))))
+         query (query-model/query {:concept-type :collection
+                                   :condition (gc/and-conds conditions)
+                                   :skip-acls? false
+                                   :page-size :unlimited
+                                   :result-format :query-specified
+                                   :result-features [:granule-counts]
+                                   :result-fields [:concept-id
+                                                   :doi
+                                                   :entry-title
+                                                   :short-name
+                                                   :version-id]})
+         result (query-exec/execute-query context query)
+         collection-concept-ids (vec (map :concept-id (:items result)))
+         my-condition (gc/and-conds (into [] (concat granule-conditions [(query-model/string-conditions :collection-concept-id collection-concept-ids true)])))
+         granule-query (query-model/query {:concept-type :granule
+                                           :condition my-condition
+                                           :page-size 0
+                                           :result-format :query-specified
+                                           :result-fields []
+                                           :aggregations {:granule-counts-by-collection-id
+                                                          {:terms {:field (q2e/query-field->elastic-field
+                                                                           :collection-concept-id :granule)
+                                                                   :size (count collection-concept-ids)}}}})
+         elastic-result (query-exec/execute-query context granule-query)
+         collection-concept-id-to-count-map (gcrf/search-results->granule-counts elastic-result)
+         {:keys [items]} result]
+     (sort-by :entry-title
+              (map #(assoc % :granule-count
+                           (get collection-concept-id-to-count-map (:concept-id %))) items)))))
 
 (defmethod collection-data :default
-  [context tag provider-id]
-  (get-collection-data context tag provider-id))
+  ([context tag provider-id]
+   (get-collection-data context tag provider-id []))
```

Comment:
Don't need the `[]`, otherwise your `get-collection-data` without conditions function would be wasted.